aopy/data/bmi3d.py

```diff
@@ -864,21 +864,31 @@ def get_kinematics(preproc_dir, subject, te_id, date, samplerate, preproc=None,
             kinematics, samplerate = preproc(raw_kinematics, samplerate)
         else:
             kinematics = raw_kinematics
+
     else:
         kinematics, samplerate = get_task_data(preproc_dir, subject, te_id, date, datatype, 
                                                samplerate, preproc=preproc, **kwargs)
 
     return kinematics, samplerate
 
 def _get_kinematic_segment(preproc_dir, subject, te_id, date, start_time, end_time, samplerate, 
-                          preproc=None, datatype='cursor', **kwargs):
+                          preproc=None, datatype='cursor', return_nan=False, **kwargs):
     '''
     Helper function to return one segment of kinematics
     '''
-    kinematics, samplerate = get_kinematics(preproc_dir, subject, te_id, date, samplerate, preproc, datatype, **kwargs)
-    assert kinematics is not None
+    if return_nan:
+        try:
+            kinematics, samplerate = get_kinematics(preproc_dir, subject, te_id, date, samplerate, preproc, datatype, **kwargs)
+            assert kinematics is not None
+
+            return get_data_segment(kinematics, start_time, end_time, samplerate), samplerate
+        except:
+            return [np.nan]
+    else: 
+        kinematics, samplerate = get_kinematics(preproc_dir, subject, te_id, date, samplerate, preproc, datatype, **kwargs)
+        assert kinematics is not None
 
-    return get_data_segment(kinematics, start_time, end_time, samplerate), samplerate
+        return get_data_segment(kinematics, start_time, end_time, samplerate), samplerate
```

Comment:
to avoid duplicating lines, i would write this as:
```suggestion
    try:
        kinematics, samplerate = get_kinematics(preproc_dir, subject, te_id, date, samplerate, preproc, datatype, **kwargs)
        assert kinematics is not None

        return get_data_segment(kinematics, start_time, end_time, samplerate), samplerate
    except Exception as e:
        if return_nan:
            return [np.nan]
        else:
            raise e
```